File: data/recomposed_projects/deepseek-coder-33b-instruct/signature/commons-graph/evosuite-test/src/test/org/apache/commons/graph/elo/DefaultKFactorBuilder_ESTest.py
from __future__ import annotations
import time
import re
import mock
import os
import unittest
import pytest
import io
import unittest
from src.main.org.apache.commons.graph.DirectedGraph import *
from src.main.org.apache.commons.graph.elo.DefaultKFactorBuilder import *

# from src.test.org.apache.commons.graph.elo.DefaultKFactorBuilder_ESTest_scaffolding import *
from src.main.org.apache.commons.graph.elo.GameResult import *
from src.main.org.apache.commons.graph.elo.PlayersRank import *
from src.main.org.apache.commons.graph.model.DirectedMutableGraph import *

# from src.main.org.evosuite.runtime.EvoAssertions import *
# from src.main.org.evosuite.runtime.EvoRunnerParameters import *
# from src.main.org.evosuite.runtime.ViolatedAssumptionAnswer import *
# from src.main.org.evosuite.shaded.org.mockito.Mockito import *
# from src.main.org.evosuite.shaded.org.mockito.stubbing.Answer import *
# from src.main.org.evosuite.shaded.org.mockito.stubbing.Stubber import *


class DefaultKFactorBuilder_ESTest(unittest.TestCase):

    def test4(self) -> None:

        directedMutableGraph0 = DirectedMutableGraph()
        defaultKFactorBuilder0 = DefaultKFactorBuilder(directedMutableGraph0, None)
        defaultKFactorBuilder0.withDefaultKFactor()

    def test2(self) -> None:

        defaultKFactorBuilder0 = DefaultKFactorBuilder(None, None)

        with pytest.raises(RuntimeError):
            defaultKFactorBuilder0.withDefaultKFactor()
            self.fail("Expecting exception: RuntimeError")

        verifyException(
            "org.apache.commons.graph.elo.DefaultKFactorBuilder", RuntimeError
        )

    def test1(self) -> None:

        directed_mutable_graph0 = DirectedMutableGraph()
        double0 = float(2324.7)
        double1 = float(-1238.378389823641)
        players_rank0 = PlayersRank()
        players_rank0.getRanking = MagicMock(
            side_effect=[double0, double1, double1, double0]
        )
        default_k_factor_builder0 = DefaultKFactorBuilder(
            directed_mutable_graph0, players_rank0
        )
        game_result0 = GameResult.DRAW
        directed_mutable_graph0.addVertex(game_result0)
        directed_mutable_graph0.addEdge(game_result0, game_result0, game_result0)
        default_k_factor_builder0.withKFactor(-1250)

    def test0(self) -> None:

        directed_mutable_graph0 = DirectedMutableGraph()
        double0 = float(0.5)
        players_rank0 = PlayersRank()
        players_rank0.getRanking = lambda x: double0
        default_k_factor_builder0 = DefaultKFactorBuilder(
            directed_mutable_graph0, players_rank0
        )
        game_result0 = GameResult()
        game_result0.DRAW = GameResult()
        directed_mutable_graph0.addVertex(game_result0.DRAW)
        game_result1 = GameResult()
        game_result1.WIN = GameResult()
        directed_mutable_graph0.addVertex(game_result1.WIN)
        directed_mutable_graph0.addEdge(
            game_result0.DRAW, game_result1.WIN, game_result1.WIN
        )
        default_k_factor_builder0.withKFactor(-1)

File: java_projects/cleaned_final_projects_evosuite/commons-graph/src/test/java/org/apache/commons/graph/elo/DefaultKFactorBuilder_ESTest.java

/*
  Licensed to the Apache Software Foundation (ASF) under one or more
  contributor license agreements.  See the NOTICE file distributed with
  this work for additional information regarding copyright ownership.
  The ASF licenses this file to You under the Apache License, Version 2.0
  (the "License"); you may not use this file except in compliance with
  the License.  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */


package org.apache.commons.graph.elo;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.shaded.org.mockito.Mockito.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.apache.commons.graph.DirectedGraph;
import org.apache.commons.graph.elo.DefaultKFactorBuilder;
import org.apache.commons.graph.elo.GameResult;
import org.apache.commons.graph.elo.PlayersRank;
import org.apache.commons.graph.model.DirectedMutableGraph;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.ViolatedAssumptionAnswer;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true) 
public class DefaultKFactorBuilder_ESTest extends DefaultKFactorBuilder_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      DirectedMutableGraph<GameResult, GameResult> directedMutableGraph0 = new DirectedMutableGraph<GameResult, GameResult>();
      Double double0 = new Double(0.5);
      PlayersRank<GameResult> playersRank0 = (PlayersRank<GameResult>) mock(PlayersRank.class, new ViolatedAssumptionAnswer());
      doReturn(double0, double0, double0, double0).when(playersRank0).getRanking(any(org.apache.commons.graph.elo.GameResult.class));
      DefaultKFactorBuilder<GameResult> defaultKFactorBuilder0 = new DefaultKFactorBuilder<GameResult>(directedMutableGraph0, playersRank0);
      GameResult gameResult0 = GameResult.DRAW;
      directedMutableGraph0.addVertex(gameResult0);
      GameResult gameResult1 = GameResult.WIN;
      directedMutableGraph0.addVertex(gameResult1);
      directedMutableGraph0.addEdge(gameResult0, gameResult1, gameResult1);
      defaultKFactorBuilder0.withKFactor((-1));
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      DirectedMutableGraph<GameResult, GameResult> directedMutableGraph0 = new DirectedMutableGraph<GameResult, GameResult>();
      Double double0 = new Double(2324.7);
      Double double1 = new Double((-1238.378389823641));
      PlayersRank<GameResult> playersRank0 = (PlayersRank<GameResult>) mock(PlayersRank.class, new ViolatedAssumptionAnswer());
      doReturn(double0, double1, double1, double0).when(playersRank0).getRanking(any(org.apache.commons.graph.elo.GameResult.class));
      DefaultKFactorBuilder<GameResult> defaultKFactorBuilder0 = new DefaultKFactorBuilder<GameResult>(directedMutableGraph0, playersRank0);
      GameResult gameResult0 = GameResult.DRAW;
      directedMutableGraph0.addVertex(gameResult0);
      directedMutableGraph0.addEdge(gameResult0, gameResult0, gameResult0);
      defaultKFactorBuilder0.withKFactor((-1250));
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      DefaultKFactorBuilder<Object> defaultKFactorBuilder0 = new DefaultKFactorBuilder<Object>((DirectedGraph<Object, GameResult>) null, (PlayersRank<Object>) null);
      // Undeclared exception!
      try { 
        defaultKFactorBuilder0.withDefaultKFactor();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.elo.DefaultKFactorBuilder", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      DirectedMutableGraph<GameResult, GameResult> directedMutableGraph0 = new DirectedMutableGraph<GameResult, GameResult>();
      GameResult gameResult0 = GameResult.WIN;
      directedMutableGraph0.addVertex(gameResult0);
      directedMutableGraph0.addEdge(gameResult0, gameResult0, gameResult0);
      PlayersRank<GameResult> playersRank0 = (PlayersRank<GameResult>) mock(PlayersRank.class, new ViolatedAssumptionAnswer());
      doReturn((Double) null).when(playersRank0).getRanking(any(org.apache.commons.graph.elo.GameResult.class));
      DefaultKFactorBuilder<GameResult> defaultKFactorBuilder0 = new DefaultKFactorBuilder<GameResult>(directedMutableGraph0, playersRank0);
      // Undeclared exception!
      try { 
        defaultKFactorBuilder0.withKFactor(1714);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.graph.elo.DefaultKFactorBuilder", e);
      }
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      DirectedMutableGraph<Object, GameResult> directedMutableGraph0 = new DirectedMutableGraph<Object, GameResult>();
      DefaultKFactorBuilder<Object> defaultKFactorBuilder0 = new DefaultKFactorBuilder<Object>(directedMutableGraph0, (PlayersRank<Object>) null);
      defaultKFactorBuilder0.withDefaultKFactor();
  }
}
